### PR TITLE
NodeJS replacement scans

### PR DESCRIPTION
**Pull request by whscullin**
Add support for replacement scans to Database objects.

### CODE REVIEW

File: tools/nodejs/src/database.cpp [hunk 1]
```diff
@@ -267,4 +274,99 @@ Napi::Value Database::Connect(const Napi::CallbackInfo &info) {
 	return Connection::constructor.New({Value()});
 }
 
+struct JSRSArgs {
+	std::string table = "";
+	std::string function = "";
+	duckdb::Value parameter;
```
Comment (Mause) [inline tools/nodejs/src/database.cpp:280]: Is there a particular reason you're limiting this to a single parameter?

**Review by Mause — CHANGES_REQUESTED** [review]
Thanks for this, looks great! Just a couple of comments

File: tools/nodejs/src/database.cpp [hunk 2]
```diff
@@ -267,4 +274,99 @@ Napi::Value Database::Connect(const Napi::CallbackInfo &info) {
...
+			    ScanReplacement, duckdb::make_unique<NodeReplacementScanData>(rs));
+		}
+	}
+	duckdb_node_rs_function_t rs;
+};
+
+Napi::Value Database::RegisterReplacementScan(const Napi::CallbackInfo &info) {
+	auto env = info.Env();
+	if (info.Length() < 1) {
+		Napi::TypeError::New(env, "Holding it wrong").ThrowAsJavaScriptException();
```
[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/src/database.cpp:354]: Heh, sure, that's just the standard error response I copied from elsewhere in the code.

File: tools/nodejs/lib/duckdb.d.ts [hunk 3]
```diff
@@ -99,6 +108,11 @@ export class Database {
   register_buffer(name: string, array: ArrowIterable, force: boolean, callback?: Callback<void>): void;
 
   unregister_buffer(name: string, callback?: Callback<void>): void;
+
+  register_replacement_scan(
```
[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/lib/duckdb.d.ts:112]: I initially did that, but I noticed that all of the other register functions used snake case so I switched for consistency.

File: tools/nodejs/lib/duckdb.d.ts [hunk 4]
```diff
@@ -99,6 +108,11 @@ export class Database {
   register_buffer(name: string, array: ArrowIterable, force: boolean, callback?: Callback<void>): void;
 
   unregister_buffer(name: string, callback?: Callback<void>): void;
+
+  register_replacement_scan(
+    replacementScan: ReplacementScanCallback,
+    callback?: Callback<void>
```
[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/lib/duckdb.d.ts:114]: Probably, again this is just consistent with the rest of the API.

File: tools/nodejs/src/database.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/src/database.cpp]: Updated.

[the remark this answers is not in the document]
Comment (Mause) [reply tools/nodejs/src/database.cpp]: I think the error message is less of a standard and more of a leftover joke from the initial prototype of the node bindings, they should probably all be removed

File: tools/nodejs/lib/duckdb.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mause) [reply tools/nodejs/lib/duckdb.d.ts:112]: Yeah, my intention is to update the other to the camel case style and have them all support promises - when I find the time 😅

File: tools/nodejs/lib/duckdb.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mause) [reply tools/nodejs/lib/duckdb.d.ts:114]: My hope is to have them all support promises to reduce callback hell, and the newer ones already do, so my preference would be to have all the new ones use promises instead

[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/lib/duckdb.d.ts]: Sure, no problem.

File: tools/nodejs/lib/duckdb.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (whscullin) [reply tools/nodejs/lib/duckdb.d.ts]: Updated.

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM. @Mause could you perhaps take a final look?

**Pull request merged**